# HW 2
## Problem 4

Just generic python setup first, including some helper functions to help with degree-radians conversion, and solving quadratic equations

In [2]:
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

In [3]:
def rad(th):
    return th * np.pi/180

def deg(th):
    return th * 180/np.pi

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

The pilot wants to fly due northwest relative to the ground ($\theta_G = 135^\circ$). The airplane flies at an airspeed of $v_A = 250$ mph at an angle $\theta_A$ to be determined. The wind is blowing in the direction $\theta_W = 20^\circ$ north of east at $v_W = 50$ mph.

I'm using the subscripts G, A, W for ground, air and wind, respectively.

Let's start by setting up some python variables for the known quantities:

In [4]:
vw = 50.
va = 250.
thg = rad(135.)
thw = rad(20.)

Looking from the ground, the airplane is flying at its airspeed while also being pushed by the wind:

$$
\vec v_G = \vec v_A + \vec v_W
$$

We can write this all out in components:
\begin{align}
v_{Gx} & = v_G\cos \theta_G = v_A \cos\theta_A + v_W \cos \theta_W\\
v_{Gy} & = v_G\sin \theta_G = v_A \sin\theta_A + v_W \sin \theta_W\\
\end{align}

These are two equations for two unknowns ($\theta_A$ and $v_G$), but they are a bit tricky to solve.
My approach is to solve the two equations for the components of $v_A$, then square them and add them together, a which point $\theta_A$ drops out because $\cos^2\theta_A + \sin^2\theta_A = 1$.

$$
\begin{align}
v_A \cos\theta_A &= v_G\cos \theta_G - v_W \cos \theta_W\\
v_A \sin\theta_A &= v_G\sin \theta_G - v_W \sin \theta_W\\
~\\
\Longrightarrow v_A^2 &= v_G^2 +v_W^2 - 2v_Gv_W(\cos\theta_G \cos\theta_W + \sin\theta_G \sin\theta_W)
\end{align}
$$

This is a quadratic equation for the unknown $v_G$:
$$
v_G^2 - 2v_W(\cos\theta_G \cos\theta_W + \sin\theta_G \sin\theta_W)v_G +v_W^2 - v_A^2 = 0
$$

This looks like $a v_G^2 + b v_G + c = 0$ with $a$, $b$, $c$ calculated in the python code below.

In [5]:
a = 1
b = - 2*vw*(cos(thg)*cos(thw) + sin(thg)*sin(thw))
c = vw**2 - va**2
a,b,c

(1, 42.26182617406994, -60000.0)

So we can use the quadratic formula to get two possible solutions for $v_G$:

In [5]:
vg_p, vg_m = quadratic(a, b, c)
vg_p, vg_m

(224.72781766681038, -266.9896438408803)

We're looking for the positive solution (it's a speed, or put differently, the plane isn't supposed to fly backwards).

In [6]:
vg = vg_p
vg

224.72781766681038

Finally, we still need the angle $\theta_A$ the plane is headed. We can get that by taking the two equations above for $v_A\cos\theta_A$ and $v_A\sin\theta_A$, dividing the 2nd equation by the first to get $\tan\theta_A$:
$$
\tan\theta_A = \frac{v_A\sin\theta_A}{v_A\cos\theta_A} = \frac{v_G\sin\theta_G - v_W\sin\theta_W}{v_G\cos\theta_G - v_W\cos\theta_W}
$$
Plugging in the numbers, taking the $\arctan$ to get the actual angle and converting to degrees,

In [7]:
tha = arctan2(vg*sin(thg) - vw*sin(thw), vg*cos(thg) - vw*cos(thw))
deg(tha)

145.4432505774823

we find that the plane's heading should be $145^\circ$ N of E, ie. $35^\circ$ N of W.